skimage/morphology/_max_tree.pyx

```diff
@@ -0,0 +1,410 @@
+"""_max_tree.pyx - building a max-tree from an image.
```

Comment:
I'm missing Cython directives like
```
#cython: cdivision=True
#cython: boundscheck=False
#cython: nonecheck=False
#cython: wraparound=False
```
Is this intentional?

Reply:
No, there was no particular reason. Thanks for the comment.

Reply:
Regarding the GIL: thanks for the suggestion; this looks very interesting. 

I must admit that I do not know anything about this, so I would prefer for now to get the PR done without this modification, unless there is a strong incentive to include this.

Reply:
It's really useful when using multithreading but otherwise it has no impact and it's easy to add later. So skipping this should be fine. If you want I can add the required modifications after this PR is done.